Warnings in building testing/db/c on GCC8

<!-- Before filling this issue, please read the wiki (https://github.com/fossology/fossology/wiki)
and search if the bug do not already exists in the issues (https://github.com/fossology/fossology/issues). -->

### Logs

```lisp
libfodbreposysconf.c: In function ‘create_db_repo_sysconf’:
libfodbreposysconf.c:180:13: warning: passing argument 1 to restrict-qualified parameter aliases with argument 3 [-Wrestrict]
     sprintf(CMD, "%s -c %s", CMD, sysconfdir);
             ^~~              ~~~
libfodbreposysconf.c:187:13: warning: passing argument 1 to restrict-qualified parameter aliases with argument 3 [-Wrestrict]
     sprintf(CMD, "%s -e", CMD);
             ^~~           ~~~
libfodbreposysconf.c: In function ‘get_repodir’:
libfodbreposysconf.c:289:11: warning: passing argument 1 to restrict-qualified parameter aliases with argument 3 [-Wrestrict]
   sprintf(RepoDir, "%stestDbRepo%s", RepoDir, get_test_name());
           ^~~~~~~                    ~~~~~~~
libfodbreposysconf.c: In function ‘create_db_repo_sysconf’:
libfodbreposysconf.c:180:21: warning: ‘ -c ’ directive writing 4 bytes into a region of size between 1 and 256 [-Wformat-overflow=]
     sprintf(CMD, "%s -c %s", CMD, sysconfdir);
                     ^~~~
libfodbreposysconf.c:180:5: note: ‘sprintf’ output 5 or more bytes (assuming 260) into a destination of size 256
     sprintf(CMD, "%s -c %s", CMD, sysconfdir);
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
libfodbreposysconf.c:187:21: warning: ‘ -e’ directive writing 3 bytes into a region of size between 1 and 256 [-Wformat-overflow=]
     sprintf(CMD, "%s -e", CMD);
                     ^~~
libfodbreposysconf.c:187:5: note: ‘sprintf’ output between 4 and 259 bytes into a destination of size 256
     sprintf(CMD, "%s -e", CMD);
     ^~~~~~~~~~~~~~~~~~~~~~~~~~
libfodbreposysconf.c: In function ‘get_repodir’:
libfodbreposysconf.c:289:23: warning: ‘testDbRepo’ directive writing 10 bytes into a region of size between 1 and 256 [-Wformat-overflow=]
   sprintf(RepoDir, "%stestDbRepo%s", RepoDir, get_test_name());
                       ^~~~~~~~~~
libfodbreposysconf.c:289:3: note: ‘sprintf’ output 11 or more bytes (assuming 266) into a destination of size 256
   sprintf(RepoDir, "%stestDbRepo%s", RepoDir, get_test_name());
   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

--- a/src/testing/db/c/libfodbreposysconf.c
+++ b/src/testing/db/c/libfodbreposysconf.c
@@ -174,19 +174,50 @@ int create_db_repo_sysconf(int type, char* agent_name, char* sysconfdir) {
   }
 #endif
 
-  char CMD[ARRAY_LENGTH] = "../../../testing/db/createTestDB.php";
-  if(sysconfdir != NULL)
+  const char INIT_CMD[] = "../../../testing/db/createTestDB.php";
+  char *CMD, *tmp;
+
+  CMD = (char *)malloc(strlen(INIT_CMD) + 1);
+  if (!CMD)
   {
-    sprintf(CMD, "%s -c %s", CMD, sysconfdir);
+    return -1;
+  }
+  sprintf(CMD, "%s", INIT_CMD);
+
+  if (sysconfdir)
+  {
+    tmp = (char *)malloc(strlen(CMD) + 4 + strlen(sysconfdir) + 1);
+    if (!tmp)
+    {
+      free(CMD);
+      return -1;
+    }
+    sprintf(tmp, "%s -c %s", CMD, sysconfdir);
+    free(CMD);
+    CMD = tmp;
   }
 
-  if (1 == type) {
-    command_output(CMD);
+  switch(type)
+  {
+    case 0:
+      tmp = (char *)malloc(strlen(CMD) + 4);
+      if (!tmp)
+      {
+        free(CMD);
+        return -1;
+      }
+      sprintf(tmp, "%s -e", CMD);
+      free(CMD);
+      CMD = tmp;
+    case 1:
+      command_output(CMD);
+      break;
+    default:
+      break;
   }
-  else if (0 == type) {
-    sprintf(CMD, "%s -e", CMD);
-    command_output(CMD);
-  }
+
+  free(CMD);
+
   int argc = 3;
   char* argv[] = {agent_name, "-c", Sysconf};
 
@@ -282,11 +313,25 @@ char* get_confFile() {
  * \return repo path
  */
 char* get_repodir() {
-  memset(RepoDir, '\0', sizeof(RepoDir));
-  strcpy(RepoDir, Sysconf);
+  strncpy(RepoDir, Sysconf, ARRAY_LENGTH);
+  RepoDir[ARRAY_LENGTH-1] = '\0';
+
   char* test_name_tmp = strstr(RepoDir, "testDbConf");
-  *test_name_tmp = 0;
-  sprintf(RepoDir, "%stestDbRepo%s", RepoDir, get_test_name());
+  if (test_name_tmp)
+  {
+    *test_name_tmp = '\0';
+  }
+
+  char *tmp = malloc(strlen(RepoDir) + 1);
+  if (!tmp) {
+    return NULL;
+  }
+
+  sprintf(tmp, "%s", RepoDir);
+  sprintf(RepoDir, "%stestDbRepo%s", tmp, get_test_name());
+
+  free(tmp);
+
 #ifdef TEST
   printf("RepoDir is:%s\n", RepoDir);
 #endif

